FoodFinder/Views/ViewControllers/MapViewController.swift

```diff
@@ -13,27 +13,79 @@ protocol MapViewControllerDelegate: class {
     func didFetchRestaurants(_ restaurants: [Restaurant])
 }
 
+class LocationManagerDelegate: NSObject, CLLocationManagerDelegate {
+    let regionRadius: CLLocationDistance = 4500
+
+    let mapView: MKMapView
+
+    init(mapView: MKMapView) {
+        self.mapView = mapView
+    }
+
+    private func createRegion(for location: CLLocation) -> MKCoordinateRegion {
+        let center = CLLocationCoordinate2D(latitude: location.coordinate.latitude,
+                                            longitude: location.coordinate.longitude)
+
+        let region = MKCoordinateRegion(center: center,
+                                        latitudinalMeters: regionRadius,
+                                        longitudinalMeters: regionRadius)
+
+        return region
+    }
+
+    func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
+        if let location = locations.first { // current user location
+            let region = createRegion(for: location)
+            mapView.setRegion(region, animated: true)
+        }
+    }
+
+    func locationManager(_ manager: CLLocationManager, didFailWithError error: Error) {
+        print("Location Manager failed with error: \(error)")
+    }
+}
```

Comment:
Ok so this `LocationManagerDelegate` now responds to the appropriate `CLLocationManagerDelegate` callbacks. The name doesn't really make sense at this point, but this takes updates from the location manager, and then updates the map with the hard coded region size. That's it. This could get annoying if you ever want to move the map around by hand, but it is for a later time.